eslint.config.mjs

```diff
@@ -0,0 +1,222 @@
+    languageOptions: {
+      globals: {
+        ...globals.node
+      }
+    }
+  },
+  {
+    name: "Globals for lib/",
+    files: ["lib/**/*.{js,ts,jsx,tsx}"],
+    languageOptions: {
+      globals: {
+        ...globals.browser,
+        ...globals.commonjs,
+        process: true
+      }
+    }
+  },
+  {
+    name: "Globals for test/",
+    files: ["test/**/*.{ts,tsx}"],
+    languageOptions: {
+      globals: {
+        ...globals.jasmine
+      }
+    }
+  },
+  {
+    name: "Shared config for everything/",
+    files: ["{buildprocess,lib}/**/*.{js,ts,jsx,tsx}", "test/**/*.{ts,tsx}"],
+    extends: [
+      pluginReact.configs.flat.recommended
+      // TODO: Enable next line when upgrading to React 17+.
+      // pluginReact.configs.flat['jsx-runtime'],
+    ],
+    plugins: {
+      "react-hooks": fixupPluginRules(pluginReactHooks)
+    },
+    languageOptions: {
+      // The React plugin's recommended config does not contain language options.
+      ...pluginReact.configs.flat.recommended.languageOptions,
+      ecmaVersion: 2019,
+      sourceType: "script",
+
+      parserOptions: {
+        projectService: true,
+        tsconfigRootDir: import.meta.dirname,
+        ecmaFeatures: {
+          legacyDecorators: true
+        }
+      }
+    },
+    settings: {
+      react: {
+        version: "detect"
+      }
+    },
+    rules: {
+      // TODO: re-enable the disabled @typescript-eslint rules.
+      "@typescript-eslint/no-explicit-any": ["off", { ignoreRestArgs: true }],
+      "@typescript-eslint/no-namespace": "off",
+      "@typescript-eslint/no-this-alias": "off",
+
+      "@typescript-eslint/no-empty-object-type": [
+        "error",
+        {
+          allowInterfaces: "with-single-extends"
+        }
+      ],
+      // The no-useless-constructor needs to be disabled for
+      // the @typescript-eslint-rule.
+      "no-useless-constructor": "off",
+      "@typescript-eslint/no-useless-constructor": "error",
+      "@typescript-eslint/unified-signatures": "error",
+      "@typescript-eslint/consistent-type-assertions": "error",
+      "react-hooks/exhaustive-deps": "error",
+
+      "react/jsx-boolean-value": [
+        "error",
+        "never",
+        {
+          always: []
+        }
+      ],
+
+      "react/no-arrow-function-lifecycle": "error",
+      "react/no-invalid-html-attribute": "error",
+      "react/jsx-no-useless-fragment": "error",
+      "react/jsx-no-constructed-context-values": "error",
+      "react/jsx-fragments": ["error", "syntax"],
+
+      "react/jsx-no-duplicate-props": [
+        "error",
+        {
+          ignoreCase: true
+        }
+      ],
+
+      "react/jsx-pascal-case": [
+        "error",
+        {
+          allowAllCaps: true,
+          ignore: []
+        }
+      ],
+
+      "react/no-danger": "warn",
+      "react/no-did-update-set-state": "error",
+      "react/no-will-update-set-state": "error",
+      "react/prop-types": "error",
```

Comment:
The `react/prop-types` was disabled in https://github.com/TerriaJS/terriajs/pull/7382#issuecomment-2653021794, but apparently doesn't trigger 1400+ warnings in ESLint 9, so re-enable the rule.